feat:added action for pokemon detail

NicoEnriquezFFuFm opened the pull request:
- added GetPokemonsDetailAction function inside pokemon_actions.dart
- added constant variable pokemonNameTest inside constants.dart
- added PokemonDetail inside app_state.dart
- modified dispatch from GetPokemonsAction to GetPokemonsDetailAction inside home_page.dart

outcome of GetPokemonsDetailAction
![SavedScreenshot_GetPokemonDetailsAction](https://user-images.githubusercontent.com/128012634/227092505-f96bb723-b1fa-48c1-acd4-98fda601864a.PNG)


flutter formatted
![image](https://user-images.githubusercontent.com/128012634/227092532-1bc4ab6b-076a-4fe5-a365-2992e454c723.png)


flutter analyzed
![flutter_analyze_detailsAction](https://user-images.githubusercontent.com/128012634/227092446-8178a436-2df0-4090-bd8f-77210f9fa91a.PNG)

NicoEnriquezFFuFm committed:
feat:added action for pokemon detail
- added GetPokemonsDetailAction function inside pokemon_actions.dart
- added constant variable pokemonNameTest inside constants.dart
- added PokemonDetail inside app_state.dart
- modified dispatch from GetPokemonsAction to GetPokemonsDetailAction inside home_page.dart

NicoEnriquezFFuFm committed:
feat:added action for pokemon detail
- modified name of GetPokemonsDetailAction to GetPokemonDetailAction

shawn-ffufm commented on lib/state/actions/pokemon_actions.dart:
```diff
@@ -12,3 +12,12 @@ class GetPokemonsAction extends ReduxAction<AppState> {
+class GetPokemonDetailAction extends ReduxAction<AppState> {
+  @override
+  Future<AppState> reduce() async {
+    final pokeDetail = await ApiService().pokemonApi.getPokemonDetails(name: pokemonNameTest);
```
name should be dynamic not fixed value, utilize constructor parameter injection instead 🙂

shawn-ffufm replied:
kindly improve the naming of pokeDetail 🙂

shawn-ffufm commented on lib/features/home_page.dart:
```diff
@@ -9,7 +9,7 @@ class HomePage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    StoreProvider.dispatch<AppState>(context, GetPokemonsAction());
+    StoreProvider.dispatch<AppState>(context, GetPokemonDetailAction());
```
kindly add a todo that this will be removed later

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -7,3 +7,4 @@ const double fontSize = 24;
 const homePage = "This is a HomePage";
 const offSet = '0';
 const limit = '151';
+const pokemonNameTest = "bulbasaur";
```
kindly add a todo comment for this

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

NicoEnriquezFFuFm committed:
feat:added action for pokemon detail
(pr feedback)
- added constructor injection to GetPokemonDetailAction
- modified name of pokeDetail to pokemonDetail
- added TODOs in constants.dart(line 11) and home_page.dart(line 12)

shawn-ffufm commented on lib/state/actions/pokemon_actions.dart:
```diff
@@ -12,3 +12,17 @@ class GetPokemonsAction extends ReduxAction<AppState> {
+class GetPokemonDetailAction extends ReduxAction<AppState> {
+  GetPokemonDetailAction({required this.pokemonNameTest});
+
+  final String pokemonNameTest;
```
name this properly please, this is not a test parameter anymore

shawn-ffufm reviewed (changes requested):
see pr feedback

NicoEnriquezFFuFm committed:
feat:added action for pokemon detail
(pr feedback)
- modified pokemonNameTest variable to pokemonName to both constants and parameters
